File: sdk/python/examples/research.py
import os

from weaver_sdk import WeaverClient


def main() -> None:
    base_url = os.getenv("WEAVER_BASE_URL", "http://127.0.0.1:8001")
    client = WeaverClient(base_url=base_url)

    for ev in client.chat_sse(
        {"messages": [{"role": "user", "content": "Give me a 3-bullet summary of Weaver."}]}
    ):
        if ev["type"] == "text":
            print(ev["data"]["content"], end="", flush=True)
        if ev["type"] == "done":
            break

    print()
    print("thread_id:", client.last_thread_id)


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/research.mjs
import { WeaverClient } from '../dist/index.js'

const baseUrl = process.env.WEAVER_BASE_URL || 'http://127.0.0.1:8001'
const client = new WeaverClient({ baseUrl })

const payload = {
  messages: [{ role: 'user', content: 'Give me a 3-bullet summary of Weaver.' }],
  stream: true,
}

for await (const ev of client.chatSse(payload)) {
  if (ev.type === 'text') process.stdout.write(String(ev.data?.content || ''))
  if (ev.type === 'done') break
}

process.stdout.write('\n')
console.log('thread_id:', client.lastThreadId)

